runtime/common/src/parachains.rs

```diff
@@ -173,6 +332,13 @@ decl_storage! {
 		///
 		/// `None` if not yet updated.
 		pub DidUpdate: Option<Vec<ParaId>>;
+
+		/// The mapping from parent block hashes to session indexes.
+		///
+		/// Used for double vote report validation.
+		/// This is not pruned at the moment.
```

Comment:
Do you have any thoughts on how it might be pruned in the future? I'd imagine we can get an API from staking/session that notifies us every time a session is removed from a bonding window (likely batches of sessions as the bonding window shrinks by era)

Reply:
What do you think about implementing a custome type `T::SessionInterface` (not it is `Self`) that would proxy calls to the already existing implementation, it looks like we need `prune_historical_up_to`. My understanding is that this will be called on every era start pruning all data up to it.

Reply:
it's not that easy: implementing a custom `SessionInterface` on `Runtime` is conflicting with implementation in substrate. I will read further into the code.

Reply:
I don't think we want to be responsible for the call to `prune_historical_up_to`, but we would want to receive notifications when that is called, and for which sessions.